Modules/cjkcodecs/multibytecodec.c

```diff
@@ -1119,6 +1197,82 @@ _multibytecodec_MultibyteIncrementalDecoder_decode_impl(MultibyteIncrementalDeco
     return NULL;
 }
 
+/*[clinic input]
+_multibytecodec.MultibyteIncrementalDecoder.getstate
+[clinic start generated code]*/
+
+static PyObject *
+_multibytecodec_MultibyteIncrementalDecoder_getstate_impl(MultibyteIncrementalDecoderObject *self)
+/*[clinic end generated code: output=255009c4713b7f82 input=4006aa49bddbaa75]*/
+{
+    PyObject *buffer;
+    unsigned long long flag;
+
+    if (self->pendingsize > 0) {
+        buffer = PyBytes_FromString((const char *)self->pending);
+    } else {
+        buffer = PyBytes_FromString("");
+    }
+    if (buffer == NULL) {
+        return NULL;
+    }
+
+    flag = self->state.ull;
+
+    return Py_BuildValue("NK", buffer, flag);
+}
+
+/*[clinic input]
+_multibytecodec.MultibyteIncrementalDecoder.setstate
+    state: object
+    /
+[clinic start generated code]*/
+
+static PyObject *
+_multibytecodec_MultibyteIncrementalDecoder_setstate(MultibyteIncrementalDecoderObject *self,
+                                                     PyObject *state)
+/*[clinic end generated code: output=26556d397fc504d8 input=1d31c7a7b0b04371]*/
+{
+    PyObject *buffer;
+    PyObject *ret;
+    Py_ssize_t buffersize;
+    char *bufferstr;
+    unsigned long long flag;
+
+    if (!PyTuple_Check(state)) {
+        PyErr_SetString(PyExc_TypeError, "state argument must be a tuple");
+        return NULL;
+    }
+    if (!PyArg_ParseTuple(state, "OK;setstate(): illegal state argument",
+                          &buffer, &flag))
```

Comment:
You can use `S` for better error message..
https://docs.python.org/3/library/codecs.html#codecs.IncrementalDecoder.getstate